src/attribute.rs

```diff
@@ -129,6 +129,73 @@ impl<'src> Display for Attribute<'src> {
   }
 }
 
+#[derive(Default, Debug, Clone, PartialEq)]
+pub(crate) struct AttributeSet<'src> {
+  inner: BTreeSet<Attribute<'src>>,
+}
+
+impl<'src> Serialize for AttributeSet<'src> {
```

Comment:
Can we avoid needing to write a custom a serialize implementation if we make the struct:

```
pub(crate) struct AttributeSet<'src>(BTreeSet<Attribute<'src>>);
```

I think in that case serde will serialize the struct as just the inner contents.